nix environment fails to build due to locale settings

I added a .envrc to my unison directory with `use flake` after installing `nix-direnv` and `direnv` packages from nix. When I enter the directory now and try to build. I DO see all the expected version of the tools, but when I try to build, it fails with the error:

```
/home/stew/devel/unison/codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs:377:32: error:
    * Exception when trying to run compile-time code:
        unison/sql/create.sql: hGetContents: invalid argument (invalid byte sequence)
      Code: template-haskell-2.18.0.0:Language.Haskell.TH.Quote.quoteExp
              hereFile "unison/sql/create.sql"
    * In the quasi-quotation: [hereFile|unison/sql/create.sql|]
    |
377 |   executeStatements (Text.pack [hereFile|unison/sql/create.sql|])
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

recoverEncode: invalid argument (invalid character)
```

@ceedubs pointed out that this is a know issue that @mitchellwrosen has encountered here: https://github.com/commercialhaskell/stack/issues/4052

Seems that we should be able to alter our flake.nix to use the workaround suggested by mitchell in the bug.

--- a/flake.nix
+++ b/flake.nix
@@ -26,6 +26,7 @@
               exports.ghc
               pkg-config
               zlib
+              glibcLocales
             ] ++ nativePackages;
             # workaround for https://gitlab.haskell.org/ghc/ghc/-/issues/11042
             shellHook = ''

